apps/mochi-web/pages/profile/transactions/index.tsx

```diff
@@ -52,14 +92,131 @@ const AppPageHeader = () => {
   )
 }
 
+const MAX_TRANSACTION_TO_DISPLAY = 30
+
 const App: NextPageWithLayout = () => {
   const { profile } = useLoginWidget()
 
+  const [filterPlatform, setFilterPlatform] = useState<
+    TransactionPlatform | 'all'
+  >('all')
+  const [filterType, setFilterType] = useState<TransactionActionType | 'all'>(
+    'all',
+  )
+
+  const { transactions: _transactions, isLoading } = useFetchProfileTransaction(
+    profile?.id ?? '',
+    Boolean(profile?.id),
+    {
+      action: ignoreOptionAll(filterType),
+      platform: ignoreOptionAll(filterPlatform),
+    },
+  )
+  const transactions = useMemo(
+    () => _transactions?.slice(0, MAX_TRANSACTION_TO_DISPLAY),
+    [_transactions],
+  )
+
+  const columns: ColumnProps<ModelProfileTransaction>[] = [
```

Comment:
Can you add `width` for each column and specify `min-width` for the container to make it more adaptable on mobile? You can refer to the implementation from the tables on the application's pages.